src/v/cluster/data_migration_worker.cc

```diff
@@ -248,16 +249,21 @@ ss::future<errc> worker::do_work(
 
     switch (sought_state) {
     case state::prepared:
-        co_return co_await flush(partition);
+        co_return co_await partition->flush_archiver();
     case state::executed: {
         auto block_res = co_await block(partition, true);
-        if (block_res != errc::success) {
-            co_return block_res;
+        if (!block_res.has_value()) {
+            co_return block_res.error();
         }
-        co_return co_await flush(partition);
+        auto block_offset = block_res.value();
+
+        auto deadline = model::timeout_clock::now() + 5s;
```

Comment:
It seems to me that this timeout value is very low considering the possible amount of work is there to be done. Should we consider making it larger and configurable ?

Reply:
It will be retried by migration reconciliation, including when controller leadership changes. Since partition is blocked already translation backlog will eventually get processed. Translator flush doesn't do any active work, it just waits for certain offset to be translated. The only problem is cloud storage flush will be invoked every time. @WillemKauf @andrwng if a partition does not receive any further writes is it much overhead to flush its cloud data every few seconds?

Reply:
With this call we are waiting for the translator to actually execute the translation work, and this may take a while depending on the translation gap. If all the calls are idempotent then this is great, it will just be retried and eventually succeed